src/CppUTest/JUnitTestOutput.cpp

```diff
@@ -151,8 +151,24 @@ void JUnitTestOutput::printCurrentTestStarted(const UtestShell& test)
 SimpleString JUnitTestOutput::createFileName(const SimpleString& group)
 {
     SimpleString fileName = "cpputest_";
+    if (!impl_->package_.isEmpty()) {
+        fileName += impl_->package_;
+        fileName += "_";
+    }
     fileName += group;
+
+    // special character list based on: https://en.wikipedia.org/wiki/Filename
```

Comment:
Could you extract the replacement in a function. Also please make it a look over a static array :)

Reply:
done